analyzer/src/steps/typing.rs

```diff
@@ -711,9 +711,12 @@ fn ascribe_call(
             convert_into_string(expr, exploration, diagnostics, state)
         })
         .collect::<Vec<_>>();
+
+    let ty = if state.local_type { EXIT_CODE } else { NOTHING };
```

Comment:
It is not up to the analyzer to check if an expression is used. Every unused return must be tracked later, as it is a concern for a lot of expressions: `while true { 1 }`, `while true { 1 + 1 }`, `while true { "here" }`, val o = 5; `while true { $o }`...

`local_type` is used only to determine if the nested expression must coerce to a same type.

Reply:
We does not have this phase yet, and `is_local_type` is clearly used for expressions whose return type is retained. for now the compiler can simply check if an expression that usually returns something is typed as UNIT to pop the value, because if useless values arent popped, the operand stack gets overflowed

Reply:
will finally move this detection in the compiler in a next pull request